### ISSUE TITLE
[BarcodeScanner] CameraX poor performance vs Camera1

### ISSUE THREAD

**Issue by garciacastano09** [googlesamples/mlkit#425]
Hi all,

First, thank you for the great sample app, it really accelerates learning.

I was running the sample app with the barcode scanner, and I noticed that the LivePreviewActivity has a much better performance than the CameraXLivePreviewActivity. I'm interested on implementing a similar barcode scanning approach, and I'm concerned about using the CameraX library, given the poor performance. Here are a couple screenshots from a Samsung Galaxy S8 running both activities against the same image.

<img src="https://i.ibb.co/d5hzsSF/Screenshot-20220113-122722-MLKit-Vision.jpg" width="40%"><img src="https://i.ibb.co/fxFcDzp/Screenshot-20220113-122658-MLKit-Vision.jpg" width="40%">

What's the reason behind the performance difference? Is it a bug in the sample app or is it a performance gap between Camera1 and CameraX implementations?

Thank you,
Jorge

**Comment by zhouyiself** [comment]
Thanks for the feedback!

Which barcode scanning dependency version you're testing (com.google.mlkit:barcode-scanning:17.0.1 ?)?

I tested them locally and the performance looks similar. (resolutions are a little bit different than yours though due to the device capability, but it shouldn't mater much)

LivePreviewActivity
![3stvZpXiEBRjwvx](https://user-images.githubusercontent.com/3314484/149600100-7d96774f-b468-4f8b-9b10-cd18293e8dd1.png)
CameraXLivePreviewActivity

![5943Ad4SBCVZzjT](https://user-images.githubusercontent.com/3314484/149600113-b63f6dce-7ea8-49d5-a97e-6db1255108d3.png)

**Comment by garciacastano09** [comment]
Yes, I'm using com.google.mlkit:barcode-scanning:17.0.1, I didn't change any dependency from the master branch. Leaving here some data on the OS I have in the device. 

![Screenshot_20220115-110541_Settings](https://user-images.githubusercontent.com/11980764/149617963-3b387725-86f4-49a9-805c-bed23108de0a.jpg)

**Comment by AlexTrotsenko** [comment]
I have tested the sample app on my Pixel 5 using the QR codes which I have found in the internet (http://qrcode.meetheed.com/qrcode_examples.php)

Unfortunately I have noticed slight, but constant performance difference between Camara1 and CamaraX api as well.
https://photos.app.goo.gl/Y4XnPyi9w32psBtm8

**Comment by peerless2012** [comment]
I have the same issue with [Vision Module].

I run the vision demo on xiaomi CC 9E
soc: Qualcomm snapdragon 665

LivePreviewActivity:
![image](https://user-images.githubusercontent.com/3116469/165452127-fc958fee-7bda-4b68-b603-c8451310ad72.png)

CameraXLivewPreviewActivity:
![image](https://user-images.githubusercontent.com/3116469/165452234-d66c05ea-f56d-483d-b460-4a3f8a9b96a5.png)

So what's wrong with CameraX?

**Comment by dreamchaser96** [comment]
Have anyone figured out what was wrong with CameraX setup?

**Comment by IceCr34m** [comment]
Any news on this ? I'm getting consistently lower performance on CameraX vs Camera1 on barcode scanning and text recognition.
